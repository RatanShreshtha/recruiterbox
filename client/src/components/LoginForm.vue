<template>
  <b-form @submit="onSubmit">
    <b-form-group
      id="input-group-1"
      label="Your Username:"
      label-for="input-1"
      description="We'll never share your username with anyone else."
    >
      <b-form-input
        id="input-1"
        v-model="username"
        type="text"
        required
        placeholder="Enter username"
      ></b-form-input>
    </b-form-group>

    <b-form-group id="input-group-2" label="Your Password:" label-for="input-2">
      <b-form-input
        id="input-2"
        v-model="password"
        type="password"
        required
        placeholder="Enter password"
      ></b-form-input>
    </b-form-group>

    <b-button class="mr-2" type="submit" variant="primary">Login</b-button>
  </b-form>
</template>
<script>
import { mapActions } from "vuex";

import { apiService } from "@/service";

export default {
  data() {
    return {
      username: "",
      password: "",
      message: null,
      error: null
    };
  },
  methods: {
    ...mapActions(["login"]),
    async onSubmit(evt) {
      evt.preventDefault();
      try {
        const credentials = {
          username: this.username,
          password: this.password
        };
        console.log(credentials);
        const resp = await apiService.post("/generate-token/", credentials);
        if (resp.data.token !== undefined) {
          const data = {
            id: resp.data.id,
            kudos: resp.data.kudos,
            username: resp.data.username,
            first_name: resp.data.first_name,
            last_name: resp.data.last_name,
            email: resp.data.email,
            token: resp.data.token
          };
          this.login(data);
        }
      } catch (error) {
        console.log("error", error);
        console.log("error", error.data);
        this.error = error.message;
      }
    }
  }
};
</script>

<style></style>
