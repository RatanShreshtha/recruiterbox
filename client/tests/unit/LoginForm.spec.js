import { shallowMount, createLocalVue } from "@vue/test-utils";

import { createRenderer } from "vue-server-renderer";

import Vuex from "vuex";
import VueRouter from "vue-router";
import { BootstrapVue, IconsPlugin } from "bootstrap-vue";
import createPersistedState from "vuex-persistedstate";

import LoginForm from "@/components/LoginForm.vue";

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueRouter);

localVue.use(BootstrapVue);
localVue.use(IconsPlugin);

describe("LoginForm.vue", () => {
  describe("User is not Logged In", () => {
    let state;
    let store;

    beforeEach(() => {
      state = {
        id: null,
        kudos: 0,
        username: null,
        first_name: null,
        last_name: null,
        email: null,
        token: null
      };

      store = new Vuex.Store({
        plugins: [createPersistedState()],
        state
      });
    });

    it("matches snapshot", () => {
      const renderer = createRenderer();

      const wrapper = shallowMount(LoginForm, {
        store,
        localVue
      });

      renderer.renderToString(wrapper.vm, (err, str) => {
        if (err) throw new Error(err);
        expect(str).toMatchSnapshot();
      });
    });
  });
});
