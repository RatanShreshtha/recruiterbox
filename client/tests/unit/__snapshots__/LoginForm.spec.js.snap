// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm.vue User is not Logged In matches snapshot 1`] = `
<b-form-stub>
  <b-form-group-stub id="input-group-1" label="Your Username:" labelFor="input-1" description="We'll never share your username with anyone else." feedbackAriaLive="assertive">
    <b-form-input-stub id="input-1" required="required" value="" placeholder="Enter username" debounce="0" type="text"></b-form-input-stub>
  </b-form-group-stub>
  <b-form-group-stub id="input-group-2" label="Your Password:" labelFor="input-2" feedbackAriaLive="assertive">
    <b-form-input-stub id="input-2" required="required" value="" placeholder="Enter password" debounce="0" type="password"></b-form-input-stub>
  </b-form-group-stub>
  <b-button-stub target="_self" event="click" routerTag="a" variant="primary" type="submit" tag="button" class="mr-2">Login</b-button-stub>
</b-form-stub>
`;
